components/Cards/src/MDCCardCollectionCell.m

```diff
@@ -0,0 +1,348 @@
+ limitations under the License.
+ */
+
+#import "MDCCardCollectionCell.h"
+
+#import "MaterialMath.h"
+#import "MaterialIcons+ic_check_circle.h"
+
+static NSString *const MDCCardCellShadowElevationsKey = @"MDCCardCellShadowElevationsKey";
+static NSString *const MDCCardCellShadowColorsKey = @"MDCCardCellShadowColorsKey";
+static NSString *const MDCCardCellBorderWidthsKey = @"MDCCardCellBorderWidthsKey";
+static NSString *const MDCCardCellBorderColorsKey = @"MDCCardCellBorderColorsKey";
+static NSString *const MDCCardCellInkViewKey = @"MDCCardCellInkViewKey";
+static NSString *const MDCCardCellSelectedImageViewKey = @"MDCCardCellSelectedImageViewKey";
+static NSString *const MDCCardCellStateKey = @"MDCCardCellStateKey";
+static NSString *const MDCCardCellSelectableKey = @"MDCCardCellSelectableKey";
+
+static const CGFloat MDCCardCellSelectedImagePadding = 8;
+static const CGFloat MDCCardCellShadowElevationNormal = 1.f;
+static const CGFloat MDCCardCellShadowElevationHighlighted = 8.f;
+static const CGFloat MDCCardCellShadowElevationSelected = 8.f;
+static const CGFloat MDCCardCellCornerRadiusDefault = 4.f;
+
+
+
+@interface MDCCardCollectionCell ()
+@property(nonatomic, strong, nullable) UIImageView *selectedImageView;
+@end
+
+@implementation MDCCardCollectionCell  {
+  NSMutableDictionary<NSNumber *, NSNumber *> *_shadowElevations;
+  NSMutableDictionary<NSNumber *, UIColor *> *_shadowColors;
+  NSMutableDictionary<NSNumber *, NSNumber *> *_borderWidths;
+  NSMutableDictionary<NSNumber *, UIColor *> *_borderColors;
+  CGPoint _lastTouch;
+}
+
++ (Class)layerClass {
+  return [MDCShadowLayer class];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _shadowElevations = [coder decodeObjectForKey:MDCCardCellShadowElevationsKey];
+    _shadowColors = [coder decodeObjectForKey:MDCCardCellShadowColorsKey];
+    _borderWidths = [coder decodeObjectForKey:MDCCardCellBorderWidthsKey];
+    _borderColors = [coder decodeObjectForKey:MDCCardCellBorderColorsKey];
+    _inkView = [coder decodeObjectForKey:MDCCardCellInkViewKey];
+    _selectedImageView = [coder decodeObjectForKey:MDCCardCellSelectedImageViewKey];
+    _state = [coder decodeIntegerForKey:MDCCardCellStateKey];
+    _selectable = [coder decodeBoolForKey:MDCCardCellSelectableKey];
+    [self commonMDCCardCollectionCellInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCCardCollectionCellInit];
+  }
+  return self;
+}
+
+- (void)commonMDCCardCollectionCellInit {
+  if (_inkView == nil) {
+    _inkView = [[MDCInkView alloc] initWithFrame:self.bounds];
+    _inkView.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+    _inkView.usesLegacyInkRipple = NO;
+    _inkView.layer.zPosition = FLT_MAX;
+    [self addSubview:self.inkView];
+  }
+
+  if (self.selectedImageView == nil) {
```

Comment:
This will break UIAppearance proxy.  In this block use _selectedImageView.

Reply:
minor: I thought it was setting the value that turns of the proxy. he is getting it.